### PR TITLE
games-fps/alephone*: add Marathon trilogy

### PR DESCRIPTION
**Pull request by Kangie**

### CODE REVIEW

**Comment by Kangie** [comment]
As always, happy to implement any feedback to improve these PRs.

I've re-used (and trimmed) the old 'bungie-marathon' license. Upstream have the rights to distribute the game data files, so as long as we continue to restrict mirroring we should be in the clear.

@thesamesam any thoughts?

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+inherit desktop
+
+DESCRIPTION="Aleph One - Marathon 2 (Durandal)"
+HOMEPAGE="http://marathon.sourceforge.net/"
+SRC_URI="https://github.com/Aleph-One-Marathon/alephone/releases/download/release-${PV}/Marathon2-${PV}-Data.zip"
+
+LICENSE="bungie-marathon"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:14]: drop

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 2]
```diff
@@ -0,0 +1,38 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+inherit desktop
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:5]: New line after EAPI

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+KEYWORDS="~amd64"
+IUSE=""
+RESTRICT="mirror"
+
+RDEPEND="games-fps/alephone"
+BDEPEND="app-arch/unzip"
+
+S="${WORKDIR}/Marathon 2"
+
+MY_NAME='durandal'
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:22]: Double quotes, please.

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+LICENSE="bungie-marathon"
+SLOT="0"
+KEYWORDS="~amd64"
+IUSE=""
+RESTRICT="mirror"
+
+RDEPEND="games-fps/alephone"
+BDEPEND="app-arch/unzip"
+
+S="${WORKDIR}/Marathon 2"
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:20]: I prefer putting S directly below (no blank new line between) SRC_URI, as it's usually directly related.

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+
+MY_NAME='durandal'
+MY_DIR="/usr/share/alephone-$MY_NAME"
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:23]: Per https://projects.gentoo.org/qa/policy-guide/ebuild-format.html#pg0101 we use ${MY_NAME} style

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+
+src_install() {
+	insinto "$MY_DIR"
+	doins -r * || die "doins failed"
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:27]: No need for die, it'll "die for us" with modern EAPIs.

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+RDEPEND="games-fps/alephone"
+BDEPEND="app-arch/unzip"
+
+S="${WORKDIR}/Marathon 2"
+
+MY_NAME='durandal'
+MY_DIR="/usr/share/alephone-$MY_NAME"
+
+src_install() {
+	insinto "$MY_DIR"
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:26]: ${}

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+	doins -r * || die "doins failed"
+
+	make_desktop_entry "alephone.sh $MY_NAME" "$DESCRIPTION"
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:29]: ${}

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+
+	# Make sure the extra dirs exist in case the user wants to add some data
+	keepdir "$MY_DIR"/{Scripts,"Physics Models",Textures,Themes}
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:32]: ${}

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+}
+
+pkg_postinst() {
+	elog "To play this scenario, run:"
+	elog "alephone.sh $MY_NAME"
```
Comment (thesamesam) [inline games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:37]: ${}

File: games-fps/alephone-durandal/metadata.xml [hunk 11]
```diff
@@ -0,0 +1,16 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<!DOCTYPE pkgmetadata SYSTEM "http://www.gentoo.org/dtd/metadata.dtd">
+<pkgmetadata>
+	<maintainer type="project">
+		<email>games@gentoo.org</email>
+		<name>Gentoo Games Project</name>
+	</maintainer>
+	<maintainer type="person">
```
Comment (thesamesam) [inline games-fps/alephone-durandal/metadata.xml:8]:
```suggestion
	<maintainer type="person" proxied="yes">
```

File: games-fps/alephone/alephone-20220115.ebuild [hunk 12]
```diff
@@ -0,0 +1,82 @@
...
+
+# ffmpeg covers most audio/video use cases and the package doesn't configure with alternatives enabled (media-libs/smpeg)
+# When resolved upstream, !ffmpeg ( media-libs/libmad media-libs/libsndfile media-libs/libvorbis media-libs/smpeg )
+# with an appropriate REQUIRED_USE may be added.
+# See https://github.com/Aleph-One-Marathon/alephone/issues/85
+RDEPEND="
+	dev-libs/boost:=
+	dev-libs/expat
+	dev-libs/zziplib:=
+	media-libs/libpng:0
```
Comment (thesamesam) [inline games-fps/alephone/alephone-20220115.ebuild:32]:
```suggestion
	media-libs/libpng
```

File: games-fps/alephone/alephone-20220115.ebuild [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+	media-libs/libsdl2
+	media-libs/sdl2-image[png]
+	media-libs/sdl2-net
+	media-libs/sdl2-ttf
+	media-video/ffmpeg:=
+	virtual/opengl
+	virtual/glu
+	sys-libs/zlib
```
Comment (thesamesam) [inline games-fps/alephone/alephone-20220115.ebuild:40]: sys-libs/zlib above virtuals

File: games-fps/alephone/alephone-20220115.ebuild [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+	alsa? ( media-libs/alsa-lib )
+	curl? ( net-misc/curl )
+	speex? ( media-libs/speex )
+"
+
+DEPEND="${RDEPEND}
+	virtual/pkgconfig
```
Comment (thesamesam) [inline games-fps/alephone/alephone-20220115.ebuild:47]: pkgconfig should be in BDEPEND

File: games-fps/alephone/alephone-20220115.ebuild [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+
+pkg_postinst() {
+	xdg_mimeinfo_database_update
+	optfeature_header "Install game data:"
+	optfeature "Marathon data files" games-fps/alephone-marathon
+	optfeature "Marathon 2 Durandal data files" games-fps/alephone-durandal
+	optfeature "Marathon: Infinity data files" games-fps/alephone-infinity
+}
+
+pkg_postrm() {
```
Comment (thesamesam) [inline games-fps/alephone/alephone-20220115.ebuild:80]: Just call xdg_pkg_postrm instead (and inherit xdg.eclass).

File: games-fps/alephone/alephone-20220115.ebuild [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+src_install() {
+	default
+	dobin "${FILESDIR}"/${PN}.sh
+	doman docs/${PN}.6
+	docinto html/
+	dodoc docs/*.html
+}
+
+pkg_postinst() {
+	xdg_mimeinfo_database_update
```
Comment (thesamesam) [inline games-fps/alephone/alephone-20220115.ebuild:73]: xdg_pkg_postinst

**Review by thesamesam — COMMENTED** [review]
Looks pretty reasonable other than some comments which apply to the other ebuilds too (not made duplicate comments, so check the others too).

I assume it actually works? 😄

File: games-fps/alephone-durandal/alephone-durandal-20220115.ebuild (hunk 5 above)
Comment (Kangie) [reply games-fps/alephone-durandal/alephone-durandal-20220115.ebuild:23]: That'd be a good one for pkgcheck / the QA bot to warn about! (Not that I shouldn't have known better here)

File: games-fps/alephone/files/alephone.sh [hunk 17]
```diff
@@ -0,0 +1,28 @@
...
+
+shift
+
+# kill ARTS, because we're just that nice
+if artsshell terminate 2> /dev/null
+then
+	sleep 2
+fi
+
+"${DIR}"/"${ALEPHONE}" "$@" "$DATA"
```
Comment (thesamesam) [inline games-fps/alephone/files/alephone.sh:28]: ${}

File: games-fps/alephone/files/alephone.sh [hunk 18]
```diff
@@ -0,0 +1,28 @@
...
+
+DIR=$(dirname "${0}")
+CMD=$(basename "${0}")
+ALEPHONE=${CMD%%.sh}
+
+if [[ -z "${1}" ]]
+then
+	echo "Usage: ${0} SCENARIO FLAGS"
+	echo "Where SCENARIO is one of:"
+	for d in /usr/share/alephone-*
```
Comment (thesamesam) [inline games-fps/alephone/files/alephone.sh:11]: It's not going to respect prefix but I guess it's not a big deal for now.

File: games-fps/alephone/alephone-9999.ebuild [hunk 19]
```diff
@@ -0,0 +1,97 @@
...
+pkg_postinst() {
+	xdg_pkg_postinst
+	optfeature_header "Install game data:"
+	optfeature "Marathon data files" games-fps/alephone-marathon
+	optfeature "Marathon 2 Durandal data files" games-fps/alephone-durandal
+	optfeature "Marathon: Infinity data files" games-fps/alephone-infinity
+}
+
+pkg_postrm() {
+	xdg_pkg_postrm
```
Comment (thesamesam) [inline games-fps/alephone/alephone-9999.ebuild:96]: You can drop this bit :)

File: games-fps/alephone/alephone-9999.ebuild [hunk 20]
```diff
@@ -0,0 +1,97 @@
...
+
+src_install() {
+	default
+	dobin "${FILESDIR}"/${PN}.sh
+	doman docs/${PN}.6
+	docinto html/
+	dodoc docs/*.html
+}
+
+pkg_preinst() {
```
Comment (thesamesam) [inline games-fps/alephone/alephone-9999.ebuild:83]: You can drop this bit :) (the phase is exported and nothing else is providing it (no other eclasses), so it gets called anyway)

File: games-fps/alephone/alephone-9999.ebuild [hunk 21]
```diff
@@ -0,0 +1,97 @@
...
+	curl? ( net-misc/curl )
+	speex? ( media-libs/speex )
+	upnp? ( net-libs/miniupnpc )
+"
+
+BDEPEND="
+	virtual/pkgconfig
+"
+
+DEPEND="
```
Comment (thesamesam) [inline games-fps/alephone/alephone-9999.ebuild:51]: I'd move DEPEND above and put BDEPEND last

File: games-fps/alephone/alephone-9999.ebuild [hunk 22]
```diff
@@ -0,0 +1,97 @@
...
+inherit autotools optfeature xdg
+
+DESCRIPTION="An enhanced version of the game engine from the classic Mac game, Marathon"
+HOMEPAGE="http://source.bungie.org/"
+if [[ ${PV} = 9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/Aleph-One-Marathon/alephone/"
+	EGIT_SUBMODULES=() # Upstream includes game data as submodules, we only want the engine
+else
+	KEYWORDS="~amd64"
```
Comment (thesamesam) [inline games-fps/alephone/alephone-9999.ebuild:14]: put KEYWORDS underneath S

File: games-fps/alephone-infinity/alephone-infinity-20220115.ebuild [hunk 23]
```diff
@@ -0,0 +1,37 @@
...
+RESTRICT="mirror"
+
+RDEPEND="games-fps/alephone"
+BDEPEND="app-arch/unzip"
+
+MY_NAME="infinity"
+MY_DIR="/usr/share/alephone-${MY_NAME}"
+
+src_install() {
+	insinto "$MY_DIR"
```
Comment (thesamesam) [inline games-fps/alephone-infinity/alephone-infinity-20220115.ebuild:25]: ${}

File: games-fps/alephone/alephone-9999.ebuild [hunk 24]
```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
```
Comment (thesamesam) [inline games-fps/alephone/alephone-9999.ebuild:4]: new line below EAPI

File: games-fps/alephone/alephone-20220115.ebuild [hunk 25]
```diff
@@ -0,0 +1,89 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
```
Comment (thesamesam) [inline games-fps/alephone/alephone-20220115.ebuild:4]: new line below EAPI

**Comment by Kangie** [comment]
The only concern that I have is that alephone probably needs some USE flags enabled on ffmpeg, likely `vorbis`, `mp3` and whatever handles standard MPEG (I'm guessing layer-2) decoding.

**Comment by thesamesam** [comment]
Thanks!

**Pull request closed without merging**